e2e/api-specs/json-rpc-coverage.js

```diff
@@ -0,0 +1,494 @@
+              await TestHelpers.delay(3000);
+            }
+
+            if (call.methodName === 'eth_signTypedData_v4') {
+              call.params[1] = JSON.stringify(call.params[1]);
+            }
+          },
+        });
+      });
+    }
+
+    // get all the confirmation calls to make and expect to pass
+    // Need this now?
+    getCalls(_, method) {
+      const calls = [];
+      const isMethodAllowed = this.only
+        ? this.only.includes(method.name)
+        : true;
+      if (isMethodAllowed) {
+        if (method.examples) {
+          // pull the first example
+          const e = method.examples[0];
+          const ex = e;
+
+          if (!ex.result) {
+            return calls;
+          }
+          const p = ex.params.map((e) => e.value);
+          const params =
+            method.paramStructure === 'by-name'
+              ? paramsToObj(p, method.params)
+              : p;
+          calls.push({
+            title: `${this.getTitle()} - with example ${ex.name}`,
+            methodName: method.name,
+            params,
+            url: '',
+            resultSchema: method.result.schema,
+            expectedResult: ex.result.value,
+          });
+        } else {
+          // naively call the method with no params
+          calls.push({
+            title: `${method.name} > confirmation rejection`,
+            methodName: method.name,
+            params: [],
+            url: '',
+            resultSchema: method.result.schema,
+          });
+        }
+      }
+      return calls;
+    }
+
+    async afterRequest(_, call) {
+      await new Promise((resolve, reject) => {
+        addToQueue({
+          name: 'afterRequest',
+          resolve,
+          reject,
+          task: async () => {
+            if (call.methodName === 'wallet_addEthereumChain') {
+              const cancelButton = await Matchers.getElementByText('Cancel');
+              await Gestures.tap(cancelButton);
+            }
+
+            if (call.methodName === 'wallet_switchEthereumChain') {
+              await TestHelpers.delay(3000);
+              const cancelButton = await Matchers.getElementByText('Cancel');
+              await Gestures.tap(cancelButton);
+            }
+
+            if (call.methodName === 'wallet_requestPermissions') {
+              await TestHelpers.delay(3000);
+              const cancelButton = await Matchers.getElementByText('Cancel');
+              await Gestures.tap(cancelButton);
+            }
+
+            if (call.methodName === 'personal_sign') {
+              await TestHelpers.delay(3000);
+              const cancelButton = await Matchers.getElementByText('Cancel');
+              await Gestures.tap(cancelButton);
+            }
+
+            if (call.methodName === 'eth_signTypedData_v4') {
+              await TestHelpers.delay(3000);
+              const cancelButton = await Matchers.getElementByText('Cancel');
```

Comment:
The SignModal page object have actions for tapping the cancel button.

```suggestion
await SigningModal.tapCancelButton();
```